### PR TITLE
delta-xds: avoid sending resource names for wildcard requests on stream reconnect

### LINKED ISSUE #16063: [delta-xds] do not populate resource_names_subscribe with existing resources on reconnection for "wildcard" XDSes
*Title*: Since LDS/CDS/SRDS are wildcard XDSes, we should not populate the resource_names_subscribe with existing resources.

*Description*:
 On reconnection, DeltaXDS packs the already subscribed resources into resource_names_subscribe, which is used to mean "new subscription interests", this is confusing on the management server side since for "wildcard XDSes"( CDS/LDS/SRDS) it expects an empty resource list 

https://github.com/envoyproxy/envoy/blob/main/source/common/config/delta_subscription_state.cc#L169-L182

We should also revisit this decision of "packing existing resources into resource_names_subscribe on reconnection" since 
the "interests of these resources" is already expressed via initial_resource_versions.

### PR DESCRIPTION
**Pull request by adisuissa**
Commit Message: delta-xds: avoid sending resource names for wildcard requests on stream reconnect

Additional Description:
Avoiding sending resource names for wildcard requests after the stream is dropped.
The resources are sent in `initial_resource_versions`, but the new request must have no resources in `resource_names_subscribe` so the server will understand this is a wildcard request.
 
Risk Level: Medium? breaks current behavior, which is incorrect.
Testing: Added unit and integration tests.
Docs Changes: Updated xds docs.
Release Notes: N/A.
Platform Specific Features: N/A.
Fixes #16063

Signed-off-by: Adi Suissa-Peleg <adip@google.com>

### CODE REVIEW

File: test/common/config/new_grpc_mux_impl_test.cc [hunk 1]
```diff
@@ -179,6 +179,90 @@ TEST_F(NewGrpcMuxImplTest, ReconnectionResetsNonceAndAcks) {
...
+          EXPECT_EQ(1, added_resources.size());
+          EXPECT_TRUE(
+              TestUtility::protoEqual(added_resources[0].get().resource(), load_assignment));
+        }));
+    // Expect a reply with the nonce.
+    expectSendMessage(type_url, {}, {}, "111");
+    grpc_mux_->onDiscoveryResponse(std::move(response), control_plane_stats_);
+  }
+  // Send another response with a different resource, but where EDS is paused.
+  auto resume_cds = grpc_mux_->pause(type_url);
```
Comment (dmitri-d) [inline test/common/config/new_grpc_mux_impl_test.cc:238]: Why did you add an update with a paused mux to this test?

**Comment by dmitri-d** [comment]
lgtm, other than a small question re: test.

File: test/common/config/new_grpc_mux_impl_test.cc (hunk 1 above)
Comment (adisuissa) [reply test/common/config/new_grpc_mux_impl_test.cc:238]: This is done to emulate a disconnect before an ack is sent from the client, and ensure that the message on reconnect does include the new resource.
I'll add some comments to this test.

**Review by stevenzzzz — APPROVED** [review]
LGTM

**Review by lizan — APPROVED** [review]

**Review by yanavlasov — APPROVED** [review]
